### ISSUE TITLE
Useless for expression not reported as an unused value

### ISSUE THREAD

**Issue by cjohansen** [clj-kondo/clj-kondo#2309]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

borkdude/brew/clj-kondo: stable 2024.03.13

**platform**

via clojure-lsp

**editor**

Emacs, clojure-lsp

**problem**

I modified a Java object in a `for`, but since it's lazy and not returned or otherwise evaluated, the code never runs. This should probably be linted as an unused value?

**repro**

```sh
(defn map->Headers [headers]
  (let [headers-obj ^Headers (Headers.)]
    (for [[k v] headers]
      (->> (cond-> v
             (not (coll? v)) vector)
           (map str)
           (.add headers-obj (name k))))
    headers-obj))
```

**expected behavior**

I would have expected clj-kondo to inform me that the `for` here is useless.
